projects/driver_controls/src/stm32f0xx/ads1015.c

```diff
@@ -0,0 +1,201 @@
+#include "ads1015.h"
+#include <status.h>
+#include <stdio.h>
+#include "ads1015_def.h"
+#include "gpio_it.h"
+
+static void prv_check_status(StatusCode code) {
+  if (!code) {
+    status_msg(code, "failed");
+  }
+}
+
+// reset the internal registers and enter a power-down state
+static StatusCode prv_reset(ADS1015Data *data) {
+  uint8_t reset = ADS1015_RESET_BYTE;
+  StatusCode code = i2c_write(data->i2c_port, ADS1015_I2C_GENERAL_CALL, &reset, sizeof(reset));
+  prv_check_status(code);
+  status_ok_or_return(code);
+  return STATUS_CODE_OK;
+}
+
+// writes to register given upper and lower bytes
+static StatusCode prv_setup_register(ADS1015Data *data, uint8_t reg, uint8_t msb, uint8_t lsb) {
+  uint8_t ads1015_setup_register[] = { reg, msb, lsb };
+  StatusCode code = i2c_write(data->i2c_port, data->i2c_addr | ADS1015_I2C_ADDRESS_GND,
+                              ads1015_setup_register, SIZEOF_ARRAY(ads1015_setup_register));
+  prv_check_status(code);
+  status_ok_or_return(code);
+  return STATUS_CODE_OK;
+}
+
+// reads the register and stores the value in the given array(2 bytes)
+static StatusCode prv_read(I2CPort i2c_port, ADS1015Address i2c_addr, uint8_t reg, uint8_t *read,
+                           size_t size) {
+  StatusCode code = i2c_write(i2c_port, i2c_addr | ADS1015_I2C_ADDRESS_GND, &reg, sizeof(reg));
+  prv_check_status(code);
+  status_ok_or_return(code);
+  code = i2c_read(i2c_port, i2c_addr | ADS1015_I2C_ADDRESS_GND, read, size);
+  prv_check_status(code);
+  status_ok_or_return(code);
+  return STATUS_CODE_OK;
+}
+
+// gets current channel based on the upper byte of config register
+static ADS1015Channel prv_get_channel(uint8_t config_register_msb) {
+  uint8_t channel = (config_register_msb >> 4) & 0x7;  // 4 <= channel <= 7
```

Comment:
I'm not a huge fan of what seems to be magic numbers - maybe this math would fit better in `ads1015_defs.h`? (Don't put any functions in the header though!)